### DISCUSSION TITLE
How to send pool message?

### DISCUSSION THREAD

**Discussion by aldinokemal** [tulir/whatsmeow#486] [WhatsApp protocol Q&A]
I saw that we have PollVoteMessage. how to use this?

**Comment by IMNascimento** [comment]
Alguem ja utilizou a criação de enquete para o usuario ?

**Reply by teichmullerhodge** [reply]
Use the builtin client method: client.BuildPollCreation(...)

It requires the name, the options, and the selectableOptionCount as parameters, full example:

		pollMessage := client.BuildPollCreation(poolName, options, 1)
		recipient := types.NewJID(phone, typeServer)
		resp, err := client.SendMessage(context.Background(), recipient, pollMessage)
		if err != nil {
			logger.Error(err.Error(), true, true, sessionId, true)
			server.InternalServerError(c, err.Error())
			return
		}

**Reply by ashuvax** [reply]
@teichmullerhodge And how can I answer the survey?

**Reply by teichmullerhodge** [reply]
> @teichmullerhodge And how can I answer the survey?

I'm trying to solve this as well, but first i need to collect the option in the pool update, currently the payload doesn't send the name of the option in the event. Do you know a way to properly collect it?

**Reply by teichmullerhodge** [reply]
> @teichmullerhodge And how can I answer the survey?

I figured it out. You need to use the builtin function client.BuildPoolVote that returns a *waE2E.Message.

pollVote, err := client.BuildPollVote(c, &pollEvt.Info, pollOptions)

The function expects a context, a types.MessageInfo and an array of strings with the desired options. Note that if 
the selectableOptionsCount field is 0 you can select multiple anwers, if is 1 you can't.

Example of my endpoint using GIN:

type SendPollVoteBody struct {
	PollMessageId string   `json:"PollMessageId"`
	SessionId     string   `json:"SessionId"`
	PollOptions   []string `json:"PollOptions"`
	Number        string   `json:"Number"`
	IsGroup       bool     `json:"IsGroup"`
}

func (MessageMediaAPI) SendPollVote(engine *gin.Engine) {
	engine.POST("/send-pollvote", func(c *gin.Context) {
		var RequestBody structs.SendPollVoteBody
		if err := c.ShouldBindJSON(&RequestBody); err != nil {
			server.BadRequest(c, err.Error())
			return
		}

		sessionId := RequestBody.SessionId
		phone := RequestBody.Number
		pollOptions := RequestBody.PollOptions
		pollMessageId := RequestBody.PollMessageId
		isGroup := RequestBody.IsGroup
		session, ok := structs.GlobalSessions[sessionId]

		if !ok {
			server.NotFound(c, server.SESSION_NOT_FOUND)
			return
		}

		if phone == "" || phone == " " {
			server.BadRequest(c, server.MISSING_PHONE)
			return
		}

		client := session.Cli
		pollEvt, err := storage.GetPollInfo(storage.StorageContainer, pollMessageId)
		if err != nil {
			logger.Error(err.Error(), true, true, sessionId, true)
			server.FailedDependency(c, "Poll info not found: "+err.Error())
			return
		}

		pollVote, err := client.BuildPollVote(c, &pollEvt.Info, pollOptions)
		if err != nil {
			logger.Error(err.Error(), true, true, sessionId, true)
			server.FailedDependency(c, err.Error())
			return
		}

		typeServer := ""
		if isGroup {
			typeServer = types.GroupServer
		} else {
			typeServer = types.DefaultUserServer
		}

		recipient := types.NewJID(phone, typeServer)
		resp, err := client.SendMessage(context.Background(), recipient, pollVote)
		if err != nil {
			logger.Error(err.Error(), true, true, sessionId, true)
			server.FailedDependency(c, err.Error())
			return
		}

		server.HttpResponse(c, 200,
			gin.H{
				"Voted":             true,
				"WhatsAppMessageId": resp.ID,
				"ServerId":          resp.ServerID,
				"WhatsAppTimestamp": resp.Timestamp,
				"Destination":       recipient,
				"Sender":            session.Cli.Store.ID,
				"MessageType":       "pollVote",
				"Channel":           "Whatsapp",
			})
	})
}
